pycloudlib/instance.py

```diff
@@ -369,7 +369,12 @@ def _wait_for_cloudinit(self, *, raise_on_failure: bool):
             then this method will raise an `OSError`.
         """
         self._log.debug('_wait_for_cloudinit to complete')
-        has_wait = "--wait" in self.execute("cloud-init status --help").stdout
+
+        result = self.execute("cloud-init status --help")
+        while result.failed:
+            result = self.execute("cloud-init status --help")
```

Comment:
This could probably use an overall timeout, so that we don't end up with, e.g., Travis jobs running indefinitely if `status --help` never succeeds.